apps/barista-design-system/src/pages/single-page/components/toc/toc.html

```diff
@@ -4,42 +4,48 @@
     (click)="_expandTocMenu()"
     [ngClass]="{ 'ba-is-active': _expandToc }"
   >
-    <a
-      href="#"
+    <p
       class="ba-toc-header ba-expandable-trigger ba-expandable-trigger-right"
       data-initialized="true"
-      >On this page</a
     >
+      On this page
+    </p>
     <div class="ba-toc-content ba-expandable-content">
       <ul class="ba-toc-list ba-list">
         <li
-          *ngFor="let heading of _headings$ | async"
+          *ngFor="let headline of tocItems"
           class="ba-toc-item"
           [class.ba-toc-item-active]="
-            _activeItems && _activeItems[0] === heading
+            _activeItems &&
+            _activeItems[0] &&
+            _activeItems[0].id === headline.id
           "
         >
-          <a
+          <span
             #headline
-            href="{{ heading.href }}"
+            [id]="headline.id"
             class="ba-toc-link"
-            (click)="_handleTocClick($event)"
-            >{{ heading.title }}</a
+            (click)="_handleTocClick(headline.id)"
           >
-          <ul *ngIf="heading.hasSubheadlines" class="ba-toc-ul">
+            {{ headline.headline }}
+          </span>
+          <ul *ngIf="headline.children" class="ba-toc-ul">
             <li
-              *ngFor="let subheading of heading.subheadlines"
+              *ngFor="let subheadline of headline.children"
               class="ba-toc-item"
               [class.ba-toc-item-active]="
-                _activeItems && _activeItems[1] === subheading
+                _activeItems &&
+                _activeItems[1] &&
+                _activeItems[1].id === headline.id
               "
             >
-              <a
+              <span
                 #headline
-                href="{{ subheading.href }}"
+                [id]="subheadline.id"
```

Comment:
Same here, there probably is already an element with that id, if you bind the id again you duplicate the id on the page.
I don't think we need the ID binding here.